tests/MongoDB.Driver.Tests/Jira/CSharp3188Tests.cs

```diff
@@ -0,0 +1,94 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Net.Sockets;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Jira
+{
+    public class CSharp3188Tests
+    {
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Ensure_that_MongoConnectionException_contains_expected_attributes([Values(false, true)] bool async)
+        {
+            RequireServer.Check().VersionGreaterThanOrEqualTo(new SemanticVersion(4, 4, 0)); // failCommand.blockTimeMS is supported since 4.4
+
+            var socketTimeout = TimeSpan.FromMilliseconds(100);
+            var serverResponseDelay = TimeSpan.FromMilliseconds(1000);
```

Comment:
Less timeout value doesn't reliably work on EG. I will try to investigate it, but it's not the scope of this particular ticket

Reply:
Rename variable and figure out why a shorter block time doesn't work (it works locally):

    var serverBlockTime = socketTimeout + TimeSpan.FromMilliseconds(100);

Reply:
I didn't fully figure out, I guess, something wrong in how the server sleep works. See this comment:https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/146#discussion_r496270569